### PR TITLE
[GCS]access task reconstruction in TaskInfoAccessor

### PR DESCRIPTION
**Pull request by micafan**
## Why are these changes needed?

Add task reconstruction methods to TaskInfoAccessor.

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Comment by micafan** [comment]
Please review this PR after [6645](https://github.com/ray-project/ray/pull/6645) merged.

File: src/ray/gcs/accessor.h [hunk 1]
```diff
@@ -232,6 +232,16 @@ class TaskInfoAccessor {
   virtual Status AsyncUnsubscribeTaskLease(const TaskID &task_id,
                                            const StatusCallback &done) = 0;
 
+  /// Updata task reconstruction to GCS asynchronously.
```
Comment (zhijunfu) [inline src/ray/gcs/accessor.h]:
```suggestion
  /// Update task reconstruction to GCS asynchronously.
```

File: src/ray/gcs/accessor.h [hunk 2]
```diff
@@ -232,6 +232,16 @@ class TaskInfoAccessor {
...
+  ///
+  /// \param data_ptr The task reconstruction that will be added to GCS.
+  /// \param callback Callback that will be called after task reconstruction
```
Comment (zhijunfu) [inline src/ray/gcs/accessor.h]:
```suggestion
  /// \param done Callback that will be called after task reconstruction
```

File: src/ray/gcs/accessor.h [hunk 3]
```diff
@@ -232,6 +232,16 @@ class TaskInfoAccessor {
   virtual Status AsyncUnsubscribeTaskLease(const TaskID &task_id,
                                            const StatusCallback &done) = 0;
 
+  /// Updata task reconstruction to GCS asynchronously.
+  ///
+  /// \param data_ptr The task reconstruction that will be added to GCS.
```
Comment (zhijunfu) [inline src/ray/gcs/accessor.h]:
```suggestion
  /// \param data_ptr The task reconstruction data that will be added to GCS.
```

File: src/ray/raylet/reconstruction_policy.cc [hunk 4]
```diff
@@ -134,21 +132,19 @@ void ReconstructionPolicy::AttemptReconstruction(const TaskID &task_id,
...
-      },
-      reconstruction_attempt));
+  RAY_CHECK_OK(gcs_client_->Tasks().AsyncUpdateTaskReconstruction(
+      reconstruction_entry,
+      /*done=*/
+      [this, task_id, required_object_id](Status status) {
+        if (status.ok()) {
+          HandleReconstructionLogAppend(task_id, required_object_id, /*success=*/true);
+        } else {
+          HandleReconstructionLogAppend(task_id, required_object_id, /*success=*/false);
```
Comment (zhijunfu) [inline src/ray/raylet/reconstruction_policy.cc]: HandleReconstructionLogAppend(task_id, required_object_id, /*success=*/status.ok());

File: src/ray/gcs/accessor.h [hunk 5]
```diff
@@ -232,6 +232,16 @@ class TaskInfoAccessor {
...
+  /// \param callback Callback that will be called after task reconstruction
+  /// has been added to GCS.
+  /// \return Status
+  virtual Status AsyncUpdateTaskReconstruction(
```
Comment (raulchen) [inline src/ray/gcs/accessor.h]: This interface is still coupled with the implementation. It'd be better to change it to something like `AttemptTaskReconstruction`. 
But considering we're going to deprecate task reconstruction table soon, this way is also fine.

**Review by raulchen — APPROVED** [review]

**Review by zhijunfu — APPROVED** [review]

File: src/ray/gcs/accessor.h (hunk 1 above)
Comment (micafan) [reply src/ray/gcs/accessor.h]: done

File: src/ray/gcs/accessor.h (hunk 3 above)
Comment (micafan) [reply src/ray/gcs/accessor.h]: done

File: src/ray/gcs/accessor.h (hunk 2 above)
Comment (micafan) [reply src/ray/gcs/accessor.h]: done

File: src/ray/gcs/accessor.h (hunk 5 above)
Comment (micafan) [reply src/ray/gcs/accessor.h]: done

**Pull request merged**